doc/SAI-Packet-Action-Group.md

```diff
@@ -0,0 +1,729 @@
+{
+   /** This member is not forwarding traffic */
+   SAI_PACKET_ACTION_GROUP_MEMBER_INACTIVE,
+
+   /** This member is active and is forwarding traffic */
+   SAI_PACKET_ACTION_GROUP_MEMBER_ACTIVE
+
+} sai_packet_action_group_member_status;
+
+typedef enum _sai_packet_action_group_member_attr_t
+{
+    /**
+     * @brief Start of attributes
+     */
+    SAI_PACKET_ACTION_GROUP_MEMBER_ATTR_START,
+
+    /**
+     * @brief Packet Action Group ID
+     *
+     * @type sai_object_id_t
+     * @flags MANDATORY_ON_CREATE | CREATE_ONLY
+     * @objects SAI_OBJECT_TYPE_PACKET_ACTION_GROUP
+     */
+    SAI_PACKET_ACTION_GROUP_MEMBER_ATTR_PACKET_ACTION_GROUP_ID = SAI_PACKET_ACTION_GROUP_MEMBER_ATTR_START,
+
+    /**
+     * @brief Packet Action Group Member Type
+     *
+     * @type sai_packet_action_group_type_t
+     * @flags MANDATORY_ON_CREATE | CREATE_ONLY
+     */
+    SAI_PACKET_ACTION_GROUP_MEMBER_ATTR_TYPE,
+
+    /**
+     * @brief Packet Action Group Action List
+     *    
+     * @type sai_packet_action_group_action_list_t pktactionlist
+     * @flags CREATE_AND_SET
+     * @default empty
+     */
+    SAI_PACKET_ACTION_GROUP_MEMBER_ATTR_ACTION_LIST,
+
+    /**
+     * @brief Packet Action Group Redirect Port/LAG interface
+     *
+     * This is the egress port/LAG interface out of which the packet needs to be
+     * sent out of the member. If this object is set as SAI_NULL_OBJECT_ID or
+     * is not specified. Then the packet needs to be dropped.
+     *
+     * @type sai_object_id_t
+     * @flags CREATE_AND_SET
+     * @default SAI_NULL_OBJECT_ID
+     */
+    SAI_PACKET_ACTION_GROUP_MEMBER_ATTR_REDIRECT_INTERFACE,
+
+    /**
+     * @brief Packet Action Group Member Weight
+     *
+     * The member’s share of the traffic processed by the group is defined by
+     * the individual member’s weight divided by the sum of the weights
+     * of all members in the group.
+     *
+     * @type sai_uint16_t
+     * @flags CREATE_AND_SET
+     * @default 1
+     * @validonly SAI_PACKET_ACTION_GROUP_MEMBER_ATTR_TYPE == SAI_PACKET_ACTION_GROUP_TYPE_SELECT
+     */
+    SAI_PACKET_ACTION_GROUP_MEMBER_ATTR_WEIGHT,
+
+    /**
+     * @brief The object to be monitored for this member.
+     *
+     * If the monitored object for the member fails, i.e. when port/lag cannot
+     * forward traffic in the data plane, the switching entity
+     * marks the failover status of the member as  
+     * SAI_PACKET_ACTION_GROUP_MEMBER_STATUS_INACTIVE and does
+     * not use it to forward traffic. If a next member exists (if any) whose monitored
+     * object in up , then the switching entity marks the failover status
+     * of the member as SAI_PACKET_ACTION_GROUP_MEMBER_STATUS_ACTIVE
+     * and it is used to forward traffic. Selection of the next active member is
+     * evaluated based on the order of creation of members.
+     *
+     * @type sai_object_id_t
+     * @flags CREATE_AND_SET
+     * @objects SAI_OBJECT_TYPE_PORT, SAI_OBJECT_TYPE_LAG
```

Comment:
Can you use bridge port instead of LAG and PORT, that way it will be valid for Port, LAG, Sub port and tunnels.